core/src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -254,7 +273,7 @@ object ConfigConvert extends LowPriorityConfigConvertImplicits {
   }
 
   // used for products
-  implicit def deriveInstanceWithLabelledGeneric[F, Repr <: HList, DefaultRepr <: HList](
+  implicit def deriveProductInstance[F, Repr <: HList, DefaultRepr <: HList](
```

Comment:
I appreciate the new names, they make sense, but I would have prefer to have the change in another PR :sweat_smile: . I think for this time is fine but I'll try to keep PRs focused on one thing in future.